common/src/main/java/org/opensearch/ml/common/connector/OciConnector.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.connector;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.ml.common.AccessMode;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.opensearch.ml.common.connector.ConnectorProtocols.OCI_SIGV1;
+
+/**
+ * Connector to OCI services
+ */
+@Log4j2
+@NoArgsConstructor
+@EqualsAndHashCode
+@org.opensearch.ml.common.annotation.Connector(OCI_SIGV1)
+public class OciConnector extends HttpConnector {
+    public static final String AUTH_TYPE_FIELD = "auth_type";
+
+    public static final String TENANT_ID_FIELD = "tenant_id";
+
+    public static final String USER_ID_FIELD = "user_id";
+
+    public static final String FINGERPRINT_FIELD = "fingerprint";
+
+    public static final String PEMFILE_PATH_FIELD = "pemfile_path";
+
+    public static final String REGION_FIELD = "region";
+
+    @Builder(builderMethodName = "ociConnectorBuilder")
+    public OciConnector(String name, String description, String version, String protocol,
+                        Map<String, String> parameters, Map<String, String> credential, List<ConnectorAction> actions,
+                        List<String> backendRoles, AccessMode accessMode, User owner) {
+        super(name, description, version, protocol, parameters, credential, actions, backendRoles, accessMode, owner);
+        validate();
+    }
+
+    public OciConnector(String protocol, XContentParser parser) throws IOException {
+        super(protocol, parser);
+        validate();
+    }
+
+
+    public OciConnector(StreamInput input) throws IOException {
+        super(input);
+        validate();
+    }
+
+    public OciConnector(String protocol, StreamInput input) throws IOException {
+        super(protocol, input);
+        validate();
+    }
+
+    private void validate() {
+        if (parameters == null) {
+            throw new IllegalArgumentException("Missing credential");
+        }
+        if (!parameters.containsKey(AUTH_TYPE_FIELD)) {
+            throw new IllegalArgumentException("Missing auth type");
+        }
+
+        final OciClientAuthType authType =
+                OciClientAuthType.from(
+                        parameters.get(AUTH_TYPE_FIELD).toUpperCase(Locale.ROOT));
+
+        if (authType == OciClientAuthType.USER_PRINCIPAL) {
```

Comment:
Do we mean we will only support `USER_PRINCIPAL` type? 

Why dont we have validations for other AuthTypes?

Reply:
I think it's because you need to validate additional fields for the user principal

Reply:
Yeah, I added a comment to make it clearer, otherwise people will be confused why other principal does not require passing any information for authentication and authorization 

User principal requires a few additional information provided through parameters. For instance principal and resource principal, all required information are provided as part of environment variables and file systems (provided when provisioning OCI compute instances) Hence, they do not require passing more parameters like user principal.